content/blog/2016/2016-08-25-sauce-pipeline.adoc

```diff
@@ -0,0 +1,242 @@
+- plugins
+- saucelabs
+- selenium
+- nightwatch
+author: lnewman
+---
+
+NOTE: This is a guest post by link:https://github.com/bitwiseman[Liam Newman],
+Technical Evangelist at Cloudbees.
+
+Testing web applications across multiple browsers on different platforms can be challenging even for smaller applications.
+With Jenkins and the
+link:https://wiki.jenkins-ci.org/display/JENKINS/Sauce+OnDemand+Plugin[Sauce OnDemand Plugin],
+you can wrangle that complexity by defining your Pipeline as Code.
+
+
+=== Pipeline &#9829; UI Testing, Too
+I recently started looking for a way to do browser UI testing for an open-source JavaScript project.
+The project is targeted primarily at
+link:https://nodejs.org[Node.js]
+but we're committed to maintaining browser-client compatibility as well.
+That means we should run tests on a matrix of browsers.
+link:https://saucelabs.com/[Sauce Labs]
+has an "open-sauce" program that provides free test instances to open-source projects.
+I decided to try using
+link:https://wiki.jenkins-ci.org/display/JENKINS/Sauce+OnDemand+Plugin[Sauce OnDemand Plugin]
+and
+link:http://nightwatchjs.org/[Nightwatch.js]
+to run Selenium tests on sample project first, before trying a full-blown suite of tests.
+
+=== Starting from Freestyle Job
+I started off by following Sauce Labs' instructions on "link:https://wiki.saucelabs.com/display/DOCS/Setting+Up+Sauce+Labs+with+Jenkins[Setting up Sauce Labs with Jenkins]"
+as far as I could.  I installed the
+link:https://wiki.jenkins-ci.org/display/JENKINS/Sauce+OnDemand+Plugin[Sauce OnDemand Plugin],
+created an account with Sauce Labs, and added my Sauce Labs credentials to Jenkins. From there I started to get a little lost.
+I'm new to Selenium and I had trouble understanding how to translate the instructions to my situation.
+I needed some working working example that I could play with.
+
+Happily, there's a whole range of samples in the github org
+link:https://github.com/saucelabs-sample-test-frameworks[saucelabs-sample-test-frameworks],
+which show how to integrate Sauce Labs with various test frameworks, including Nightwatch.js.
+I forked the Nightwatch.js sample to
+link:https://github.com/bitwiseman/JS-Nightwatch.js[bitwiseman/JS-Nightwatch.js]
+and started digging.
+
+[NOTE]
+--
+You can view the commits for this blog post in the
+link:https://github.com/bitwiseman/JS-Nightwatch.js/tree/sauce-pipeline[sauce-pipeline]
+branch of the
+link:https://github.com/bitwiseman/JS-Nightwatch.js[bitwiseman/JS-Nightwatch.js]
+repository.
+--
+
+I noticed that Nightwatch.js includes ANSI color output by default and supports JUnit result logging, so I installed the
+link:https://wiki.jenkins-ci.org/display/JENKINS/AnsiColor+Plugin[AnsiColor Plugin]
+and the
+link:https://wiki.jenkins-ci.org/display/JENKINS/JUnit+Plugin[JUnit Plugin], too.
+
+Between the sample and the rest Sauce Labs instructions, I was able to configure a Freestyle job that ran 5 tests on one browser via Sauce Connect.
```

Comment:
"rest of the"

Also, most style guides suggest that you spell out numbers smaller than ten, so s/5/five/ if you want to be strict.